src/Bridge/Elasticsearch/DataProvider/Extension/SortExtension.php

```diff
@@ -0,0 +1,88 @@
+<?php
+
+/*
+ * This file is part of the API Platform project.
+ *
+ * (c) Kévin Dunglas <dunglas@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace ApiPlatform\Core\Bridge\Elasticsearch\DataProvider\Extension;
+
+use ApiPlatform\Core\Api\IdentifiersExtractorInterface;
+use ApiPlatform\Core\Api\ResourceClassResolverInterface;
+use ApiPlatform\Core\Bridge\Elasticsearch\Util\FieldDatatypeTrait;
+use ApiPlatform\Core\Metadata\Property\Factory\PropertyMetadataFactoryInterface;
+use ApiPlatform\Core\Metadata\Resource\Factory\ResourceMetadataFactoryInterface;
+
+/**
+ * Applies selected sorting while querying resource collection.
+ *
+ * @see https://www.elastic.co/guide/en/elasticsearch/guide/current/_sorting.html
+ * @see https://www.elastic.co/guide/en/elasticsearch/reference/current/search-request-sort.html
+ *
+ * @experimental
+ *
+ * @author Baptiste Meyer <baptiste.meyer@gmail.com>
+ */
+final class SortExtension implements FullBodySearchCollectionExtensionInterface
+{
+    use FieldDatatypeTrait;
+
+    private $defaultDirection;
+    private $identifiersExtractor;
+    private $resourceMetadataFactory;
+
+    public function __construct(ResourceMetadataFactoryInterface $resourceMetadataFactory, IdentifiersExtractorInterface $identifiersExtractor, PropertyMetadataFactoryInterface $propertyMetadataFactory, ResourceClassResolverInterface $resourceClassResolver, string $defaultDirection = null)
+    {
+        $this->resourceMetadataFactory = $resourceMetadataFactory;
+        $this->identifiersExtractor = $identifiersExtractor;
+        $this->propertyMetadataFactory = $propertyMetadataFactory;
+        $this->identifiersExtractor = $identifiersExtractor;
+        $this->defaultDirection = $defaultDirection;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function applyToCollection(array &$requestBody, string $resourceClass, string $operationName = null, array $context): void
+    {
+        $orders = [];
+
+        if (null !== ($defaultOrder = $this->resourceMetadataFactory->create($resourceClass)->getAttribute('order')) && is_array($defaultOrder)) {
+            foreach ($defaultOrder as $property => $direction) {
+                if (\is_int($property)) {
+                    $property = $direction;
+                    $direction = 'asc';
+                }
+
+                $orders[] = $this->getOrder($resourceClass, $property, $direction);
+            }
+        } elseif (null !== $this->defaultDirection) {
+            foreach ($this->identifiersExtractor->getIdentifiersFromResourceClass($resourceClass) as $identifier) {
```

Comment:
You are assuming that an "identifier" property is always indexed. As, each document, in elasticsearch has the attribute `_id`, you can not be sure it is true.
The following use case fails : no "identifier" property indexed and get a collection with the default filter.  

You should not apply the default condition and let elasticsearch retrieve the collection sorted by `_id` 

However, as indicated in the ElasticSearch documentation, it is better to index the "identifier" field and duplicate the identifier content 
(see https://www.elastic.co/guide/en/elasticsearch/reference/6.3/mapping-id-field.html#CO257-1)

Reply:
> The following use case fails : no "identifier" property indexed and get a collection with the default filter.

IMHO that's by design, you get the `Composite identifiers not supported.` exception right...

Currently it's consistent with the doctrine bridge: apply default order on ID fields.